spock-core/src/main/java/org/spockframework/runtime/SpockNode.java

```diff
@@ -126,4 +130,34 @@ protected static ExecutionMode toExecutionMode(org.spockframework.runtime.model.
       .map(er -> new ExclusiveResource(er.getKey(), toLockMode(er.getMode())))
       .collect(Collectors.toSet());
   }
+
+  @Override
+  public Set<TestTag> getTags() {
+    // Lazy initialisation
+    if (tags == null) {
+      tags = new HashSet<>();
+      SpecInfo specInfo;
+      // Node info can be either FeatureInfo or SpecInfo
+      if (getNodeInfo() instanceof FeatureInfo) {
+        FeatureInfo featureInfo = (FeatureInfo) getNodeInfo();
+        addTestTagsFrom(featureInfo.getTags());
+        specInfo = featureInfo.getSpec();
+      }
+      else {
+        specInfo = (SpecInfo) getNodeInfo();
+      }
+      while (specInfo != null) {
+        addTestTagsFrom(specInfo.getTags());
+        specInfo = specInfo.getSuperSpec();
+      }
+    }
+    return tags;
+  }
+
+  private void addTestTagsFrom(List<Tag> specElementTags) {
+    specElementTags.stream()
+      .filter(tag -> tag.getKey().equals(TagExtension.TAG_EXTENSION_KEY))
+      .map(tag -> TestTag.create(tag.getName()))
```

Comment:
Does mapping twice not have any performance impacts? It certainly looks nicer. As for `getName` vs. `getValue`, I use `getName` because it directly returns a `String` and I can avoid a cast from `Object`. You can see that I always assign an identical name and value to each tag anyway. The available constructors required me to select one in which I had to assign a value, even though technically I do not need it. But maybe that is a good thing. I do not know, if reporting extensions might complain for tags without values.

Reply:
OK, I am using `.map(Tag::getName).map(TestTag::create)` now in https://github.com/spockframework/spock/pull/1437/commits/06954806aa5e552e60d1d4347011de23b5cde925. I stuck with `getName` for the reasons explained above.